### PR TITLE
Japan parser update

### PR DESCRIPTION
**Pull request by AnyaLindstromBattle**
Updated Japan parser to be come comprehensive, namely:
1.confirmedPatient is true or false, we only want to import the true cases.
2. sourceURL and deathSourceURL should be put as extra sources url in caseReferences.additionalSources
3. deceasedDate should be an event
4. patientStatus should be parsed as well

### CODE REVIEW

File: ingestion/functions/parsing/japan/japan.py
Comment (attwad) [inline ingestion/functions/parsing/japan/japan.py:1]: remove comment please

File: ingestion/functions/parsing/japan/japan.py
Comment (attwad) [inline ingestion/functions/parsing/japan/japan.py:65]: the else part just returning None is not needed

File: ingestion/functions/parsing/japan/japan.py
Comment (attwad) [inline ingestion/functions/parsing/japan/japan.py:73]: If you have both source and death urls the death one will overwrite the source one.
Consider rewriting this method so it's simpler:

```
sources = []
if "sourceURL" in additional_sourceURL:
  sources.append({"sourceUrl": additional_sourceURL["sourceURL"])
if "deathSourceURL" in additional_sourceURL:
  sources.append({"sourceUrl": additional_sourceURL["deathSourceURL"])
return sources or None
```

File: ingestion/functions/parsing/japan/japan.py
Comment (attwad) [inline ingestion/functions/parsing/japan/japan.py:117]: why always put death date here if outcome can be unknown?

File: ingestion/functions/parsing/japan/japan.py
Comment (attwad) [inline ingestion/functions/parsing/japan/japan.py:128]: `== True` is not needed

File: ingestion/functions/parsing/japan/japan_test.py
Comment (attwad) [inline ingestion/functions/parsing/japan/japan_test.py:35]: as per my other comment, you should'nt put a dateRange with null start and end fields if the outcome is unknown

File: ingestion/functions/parsing/japan/japan.py
Comment (axmb) [inline ingestion/functions/parsing/japan/japan.py:58]: Isn't this `Dict`, rather than `str`? Likewise for `convert_outcome` and `convert_date`.

https://docs.python.org/3/library/typing.html

Comment (axmb) [inline ingestion/functions/parsing/japan/japan.py:58]: Total nit, but would be good to consistently use snake_case here (JSON fields which we don't control aside, of course). So `convert_additional_sources` and `additional_source_url`, e.g..

File: ingestion/functions/parsing/japan/japan.py
Comment (axmb) [inline ingestion/functions/parsing/japan/japan.py:84]: Can make this a little simpler to follow by deleting the first else (79-81) and removing the `else:` part of the second block (82).

**Review by axmb — APPROVED** [review]
Nice! Good catch re: confirmedPatient.

File: ingestion/functions/parsing/japan/japan.py
Comment (AnyaLindstromBattle) [reply ingestion/functions/parsing/japan/japan.py:58]: sorry yes, copy pasted some functions and forgot to change it... Thanks!

**Review by attwad — APPROVED** [review]
lgtm, modulo ab's comments being addressed

File: ingestion/functions/parsing/japan/japan.py
Comment (AnyaLindstromBattle) [reply ingestion/functions/parsing/japan/japan.py:58]: Isn't the input a str here?

Comment (attwad) [reply ingestion/functions/parsing/japan/japan.py:58]: No it's a dict

**Comment by z023** [comment]
There are currently 729 cases coming through as Indonesia. ~70% of the IDs are duplicated. The ingestion has resulted in all their locations being set as a small place in Indonesia called 'Japan'. They were all identified in airport quarantine or are quarantine officers.

**Comment by z023** [comment]
The 2 cases coming through as Missouri USA were also identified in airport quarantine and their residence is unknown.

File: ingestion/functions/parsing/japan/japan.py
Comment (calremmel) [inline ingestion/functions/parsing/japan/japan.py:56]: If you wanted, you could have the function accept entry["dateAnnounced"] as a string, rather than passing in the whole entry. That way, you could reuse this function to process death_date rather than having to repeat the logic. It looks like it works fine as is, though.

File: ingestion/functions/parsing/japan/japan.py
Comment (calremmel) [inline ingestion/functions/parsing/japan/japan.py:74]: As before, you could probably fix it so that convert_date() handles this as well. Up to you!

**Comment by z023** [comment]
There are still cases coming through as Indonesia. @AnyaLindstromBattle this might be something to discuss with Felix as he sorted the geocoding for Argentina when cases were coming through as Mexico.

**Comment by AnyaLindstromBattle** [comment]
@z023 thanks for the heads up. How many cases are coming through as Indonesia? 

I believe this is partly what @attwad and @axmb were trying to sort out in #1278. i.e. what the change in that PR will allow is to specify specific cases which should be ignored (the Indonesia ones in this case). However this hasn't been merged as they were holding off until Japan was backfilled, has that been done? If so we can merge that PR and then I can change the parser accordingly.

**Comment by z023** [comment]
There are currently 737 that have come through as Indonesia (see line list download below). As Japan is not too large the parser is able to ingest the whole dataset each day so no backfilling was needed.
[2020.11.02 Japan_Source_Indonesia_Cases.xlsx](https://github.com/globaldothealth/list/files/5476291/2020.11.02.Japan_Source_Indonesia_Cases.xlsx)

**Comment by AnyaLindstromBattle** [comment]
Nice one, will merge that PR then. Ok as there are quite a few cases I'll have another look and chat to Felix about it as you suggested.

**Pull request merged**
